### ISSUE TITLE
You passed invalid parameters to 'cocoapods'

### ISSUE THREAD

**Issue by moguncool** [fastlane/fastlane#12924]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Fastfile

        
    class Fastfile: LaneFile {
     func customLane() {
            let itcTeamId = "32942222"
            let username = "iosapp@example.com"
            let appIdentifier = "com.softence.Project-G"
            let appleId = "iosapp@example.com"
            let devPortalTeamId = "5PN99XXXXX"
            let scheme = "Project-G"
            let xcodeproj = "Project-G.xcodeproj"
        let versionNumber = getVersionNumber(xcodeproj: xcodeproj)
        
        cocoapods(
            clean: true,
            podfile: "./Podfile",
            tryRepoUpdateOnError: true
        )
        
        latestTestflightBuildNumber(
            appIdentifier: appIdentifier,
            username: username,
            version: versionNumber,
            teamId: itcTeamId
        )
    
        let fastlaneContext = laneContext()
        
        let currentBuildNumber = fastlaneContext["LATEST_TESTFLIGHT_BUILD_NUMBER"] as! Int
        
        // Incremeting build number
        let newBuildNumber: Int = currentBuildNumber + 1
        
        incrementBuildNumber(
            buildNumber: String(newBuildNumber),
            xcodeproj: xcodeproj
        )
        
        buildIosApp(scheme: scheme)
        
        uploadToTestflight(
            username: username,
            appIdentifier: appIdentifier,
            appleId: appleId,
            teamId: itcTeamId,
            devPortalTeamId: devPortalTeamId
        )
	}

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>

## Terminal Output
[20:10:33]: -----------------------
[20:10:33]: --- Step: cocoapods ---
[20:10:33]: -----------------------
[20:10:33]: You passed invalid parameters to 'cocoapods'.
[20:10:33]: Check out the error below and available options by running `fastlane action cocoapods`
+----------------+-----+
|     Lane Context     |
+----------------+-----+
| VERSION_NUMBER | 1.0 |
+----------------+-----+
[20:10:33]: Unresolved conflict between options: 'repo_update' and 'try_repo_update_on_error'

+------+--------------------+-------------+
|            fastlane summary             |
+------+--------------------+-------------+
| Step | Action             | Time (in s) |
+------+--------------------+-------------+
| 1    | get_version_number | 0           |
| 💥   | cocoapods          | 0           |
+------+--------------------+-------------+

[20:10:33]: fastlane finished with errors

[!] Unresolved conflict between options: 'repo_update' and 'try_repo_update_on_error'
</pre>
</details>

### Environment

<details>
  <pre>
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.13.3                                                        |
| Ruby                        | 2.3.3                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.15.0                                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.13.3 (17D47)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib |
| OpenSSL Version             | LibreSSL 2.2.7                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 9.4.1                                                          |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile.swift`</summary>

```ruby
// This file contains the fastlane.tools configuration
// You can find the documentation at https://docs.fastlane.tools
//
// For a list of all available actions, check out
//
//     https://docs.fastlane.tools/actions
//

import Foundation

class Fastfile: LaneFile {
	func customLane() {
        let itcTeamId = "39941800"
        let username = "iosapp@appudvikleren.dk"
        let appIdentifier = "com.softence.Project-G"
        let appleId = "iosapp@appudvikleren.dk"
        let devPortalTeamId = "5PN99X4SRN"
        let scheme = "Project-G"
        let xcodeproj = "Project-G.xcodeproj"
        
        let versionNumber = getVersionNumber(xcodeproj: xcodeproj)
        
        cocoapods(
            clean: true,
            podfile: "./Podfile",
            tryRepoUpdateOnError: true
        )
        
        latestTestflightBuildNumber(
            appIdentifier: appIdentifier,
            username: username,
            version: versionNumber,
            teamId: itcTeamId
        )
    
        let fastlaneContext = laneContext()
        
        let currentBuildNumber = fastlaneContext["LATEST_TESTFLIGHT_BUILD_NUMBER"] as! Int
        
        // Incremeting build number
        let newBuildNumber: Int = currentBuildNumber + 1
        
        incrementBuildNumber(
            buildNumber: String(newBuildNumber),
            xcodeproj: xcodeproj
        )
        
        buildIosApp(scheme: scheme)
        
        uploadToTestflight(
            username: username,
            appIdentifier: appIdentifier,
            appleId: appleId,
            teamId: itcTeamId,
            devPortalTeamId: devPortalTeamId
        )
	}
}

```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.99.1  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.2        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.5.9        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.4.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.4.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.62.0       |
| faraday                   | 0.15.2       |
| unf_ext                   | 0.0.7.5      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20180417 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.3        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.13.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.5.0        |
| bundler                   | 1.16.2       |
| faraday_middleware        | 0.12.2       |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| logging                   | 2.2.2        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.21.2       |
| emoji_regex               | 0.1.1        |
| tty-cursor                | 0.6.0        |
| tty-screen                | 0.6.5        |
| json                      | 2.1.0        |
| io-console                | 0.4.5        |
| CFPropertyList            | 3.0.0        |
| libxml-ruby               | 2.9.0        |
</details>

*generated on:* **2018-07-18**
</details>
</pre>
</details>

**Comment by deathByLiche** [comment]
I'm having the same issue. It looks likes it might be a regression from this [change](https://github.com/fastlane/fastlane/commit/e10fa1f335843795bde8f20da64f20fa0943fb7b#diff-9dffaf555dca05a3ab27f2b64e01b380). Works as expected on Fastlane 2.98.0

**Comment by KrauseFx** [comment]
@deathByLiche are you using the Swift version of fastlane (`Fastfile.swift`) or Ruby based? This might be related to how we handle default values for some parameters in the Swift version

**Comment by deathByLiche** [comment]
Yeah I'm using the swift version

**Comment by moguncool** [comment]
Swift Version

**Comment by itlakshmanan** [comment]
@KrauseFx I'm also facing this same issue. Kindly give solution for this.

**Comment by deathByLiche** [comment]
@KrauseFx  After messing around and digging through the fastlane code, I think I've found the issue, though I'm not sure on what the best resolution should be. It looks like when ruby is trying to [verify_conflicts](https://github.com/fastlane/fastlane/blob/master/fastlane_core/lib/fastlane_core/configuration/configuration.rb), it is only checking if the conflicting_options keys exist in the command, it doesn't check the value of those options. 

For the cocoapods command, repo_update and try_repo_update_on_error are conflicting options. However in the Swift version, repo_update has a default value of false, and after the [change](https://github.com/fastlane/fastlane/commit/e10fa1f335843795bde8f20da64f20fa0943fb7b#diff-9dffaf555dca05a3ab27f2b64e01b380) in Fastlane 2.99.0, try_repo_updatte_on_error was changed to also have a default value of false, instead of defaulting to nil. Both these keys then show up in the command that is passed to the ruby command, regardless of their value, and the ruby command evaluates that they are conflicting options. 

I was able to reproduce this same issue on Fastlane 2.98.0 by calling cocoapods(tryRepoUpdateOnError: true)
